docs/wiki/EL/el-architecture.md

```diff
@@ -112,15 +112,32 @@ The beacon chain  invokes the new payload function and transfers the execution
 
 ##### Geth
 
-TODO: STF code links and walk through in Geth
+###### Transaction Execution in Geth
+Geth, like other Ethereum execution clients, processes transactions by verifying signatures, checking nonces, deducting gas fees, and updating the state. Transactions first enter the mempool, where they wait to be included in a block. Once picked up, Geth executes them, modifying account balances, contract storage, and other state data.
 
-Check week 2 talk by lightclient for an overview.
+🔗[ Transaction Execution Code](https://github.com/ethereum/execution-specs/blob/0f9e4345b60d36c23fffaa69f70cf9cdb975f4ba/src/ethereum/shanghai/fork.py#L542)
```

Comment:
Make it clear it's not geth specific but points to el specs. Also the whole section could be renamed and doesn't have to be focused on geth, could be just an 'EL client' instead
```suggestion
🔗[ Transaction Execution Specs Code](https://github.com/ethereum/execution-specs/blob/0f9e4345b60d36c23fffaa69f70cf9cdb975f4ba/src/ethereum/shanghai/fork.py#L542)
```